go/client/logclient.go

```diff
@@ -319,10 +319,12 @@ func (c *LogClient) GetSTH() (sth *ct.SignedTreeHead, err error) {
 }
 
 // GetProofByHash returns an audit path for the hash of an SCT.
-func (c *LogClient) GetProofByHash(hash []byte, treeSize uint64) (GetProofByHashResponse, error) {
-	var resp GetProofByHashResponse
+func (c *LogClient) GetProofByHash(ctx context.Context, hash []byte, treeSize uint64) (*GetProofByHashResponse, error) {
+	resp := new(GetProofByHashResponse)
```

Comment:
Oops, you've seen my comment in #1285? I like `var` better than `new`: it seems there's no case where `new` does anything that you can't do with `var` and `&`, so I'd rather have just one way of doing things, for consistency, and let the compiler do the escape analysis...

And either way, I'd still want it declared "at the last moment". :wink: